storage/connectors/bigquery/src/main/java/feast/storage/connectors/bigquery/writer/BigQueryFeatureSink.java

```diff
@@ -16,61 +16,99 @@
 import feast.proto.types.FeatureRowProto;
 import feast.storage.api.writer.FeatureSink;
 import feast.storage.api.writer.WriteResult;
-import feast.storage.connectors.bigquery.common.TypeUtil;
-import java.util.ArrayList;
-import java.util.List;
 import java.util.Map;
-import org.apache.beam.repackaged.core.org.apache.commons.lang3.tuple.Pair;
+import javax.annotation.Nullable;
+import org.apache.beam.sdk.coders.KvCoder;
+import org.apache.beam.sdk.coders.StringUtf8Coder;
+import org.apache.beam.sdk.io.gcp.bigquery.BigQueryServices;
+import org.apache.beam.sdk.options.ValueProvider;
 import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.transforms.View;
+import org.apache.beam.sdk.values.KV;
 import org.apache.beam.sdk.values.PCollection;
-import org.slf4j.Logger;
+import org.apache.beam.sdk.values.PCollectionView;
+import org.joda.time.Duration;
 
 @AutoValue
 public abstract class BigQueryFeatureSink implements FeatureSink {
-  private static final Logger log = org.slf4j.LoggerFactory.getLogger(BigQueryFeatureSink.class);
-
-  // Column description for reserved fields
-  public static final String BIGQUERY_EVENT_TIMESTAMP_FIELD_DESCRIPTION =
-      "Event time for the FeatureRow";
-  public static final String BIGQUERY_CREATED_TIMESTAMP_FIELD_DESCRIPTION =
-      "Processing time of the FeatureRow ingestion in Feast\"";
-  public static final String BIGQUERY_INGESTION_ID_FIELD_DESCRIPTION =
-      "Unique id identifying groups of rows that have been ingested together";
-  public static final String BIGQUERY_JOB_ID_FIELD_DESCRIPTION =
-      "Feast import job ID for the FeatureRow";
-
   public abstract String getProjectId();
 
   public abstract String getDatasetId();
 
-  public abstract BigQuery getBigQuery();
+  public abstract Duration getTriggeringFrequency();
+
+  @Nullable
+  public abstract BigQueryServices getBigQueryServices();
+
+  @Nullable
+  public abstract ValueProvider<BigQuery> getBigQueryOptions();
+
+  public abstract PCollection<KV<String, FeatureSetProto.FeatureSetSpec>> getFeatureSetSpecs();
 
   /**
-   * Initialize a {@link BigQueryFeatureSink.Builder} from a {@link BigQueryConfig}. This method
-   * initializes a {@link BigQuery} client with default options. Use the builder method to inject
-   * your own client.
+   * Mostly for internal use in tests Same as public version but allows to specify bigQuery clients
+   * explicitly
    *
    * @param config {@link BigQueryConfig}
    * @param featureSetSpecs
-   * @return {@link BigQueryFeatureSink.Builder}
+   * @param bigQueryServices {@link BigQueryServices} - internal beam BigQuery wrapper
+   * @param bigQueryOptions {@link ValueProvider} value provider of BigQuery - from gcloud sdk
+   * @return
    */
-  public static FeatureSink fromConfig(
-      BigQueryConfig config, Map<String, FeatureSetProto.FeatureSetSpec> featureSetSpecs) {
+  static FeatureSink fromConfig(
```

Comment:
Would it be possible to use the `Builder` directly instead of adding this `fromConfig`?